lib/engines/hcktest/tests.rb

```diff
@@ -433,12 +459,13 @@ def load_clients_system_info
       build_system_info(support_sysinfo)
     end
 
-    def run(tests)
+    def run(tests) # rubocop:disable Metrics/AbcSize
```

Comment:
This method indeed looks cluttered now so let's refactor it. I have two ideas:
- You can move `@total = @tests.count` to `tests_stats` and convert it into a local variable.
- `@last_done = [] if @last_done.nil?` can also be moved to `initialize`. It also makes `if @last_done.nil?` unnecessary.